src/core/lib.lua

```diff
@@ -686,6 +686,38 @@ function getenv (name)
    else return nil end
 end
 
+function fstruct(def)
```

Comment:
My inner Max cringes at the thought of more undocumented functions here :)  To me, it would seem an easier merge to not add these to core.lib.